src/utilities.jl

```diff
@@ -13,23 +13,71 @@ Returns the number of samples present in the span `aligned`.
 n_samples(aligned::AlignedSpan) = length(indices(aligned))
 
 """
-    consecutive_subspans(span::AlignedSpan, duration::Period)
+    consecutive_subspans(span::AlignedSpan, duration::Period; keep_last=true)
+    consecutive_subspans(span::AlignedSpan, n::Int; keep_last=true)
 
 Creates an iterator of `AlignedSpan` such that each `AlignedSpan` has consecutive indices
-which cover all of the original `span`'s indices. In particular,
+which cover the original `span`'s indices (when `keep_last=true`).
 
-* Each span has `n = n_samples(span.sample_rate, duration)` samples, except possibly
+* If `keep_last=true` (the default behavior), then the last span may have fewer samples than the others, and
+    * Each span has `n` samples (which is calculated as `n_samples(span.sample_rate, duration)` if `duration::Period` is supplied), except possibly
 the last one, which may have fewer.
-* The number of subspans is given by `cld(n_samples(span), n)`
-* The number of samples in the last subspan is `r = rem(n_samples(span), n)` unless `r=0`, in which
+    * The number of subspans is given by `cld(n_samples(span), n)`,
+    * The number of samples in the last subspan is `r = rem(n_samples(span), n)` unless `r=0`, in which
   case the the last subspan has the same number of samples as the rest, namely `n`.
+    * All of the indices of `span` are guaranteed to be covered by exactly one subspan
+* If `keep_last=false`, then all spans will have the same number of samples:
+    * Each span has `n` samples (which is calculated as `n_samples(span.sample_rate, duration)` if `duration::Period` is supplied)
+    * The number of subspans is given by `fld(n_samples(span), n)`
+    * The last part of the `span`'s indices may not be covered (when we can't fit in another subspan of length `n`)
 """
-function consecutive_subspans(span::AlignedSpan, duration::Period)
+function consecutive_subspans(span::AlignedSpan, duration::Period; keep_last=true)
     n = n_samples(span.sample_rate, duration)
-    return consecutive_subspans(span::AlignedSpan, n)
+    return consecutive_subspans(span::AlignedSpan, n; keep_last)
 end
 
-function consecutive_subspans(span::AlignedSpan, n::Int)
+function consecutive_subspans(span::AlignedSpan, n::Int; keep_last=true)
     index_groups = Iterators.partition((span.first_index):(span.last_index), n)
+    if !keep_last
+        r = rem(n_samples(span), n)
+        if r != 0
+            # Drop the last element
+            grps = Iterators.take(index_groups, fld(n_samples(span), n))
+            return (AlignedSpan(span.sample_rate, first(I), last(I)) for I in grps)
+        end
+    end
     return (AlignedSpan(span.sample_rate, first(I), last(I)) for I in index_groups)
 end
+
+"""
+    consecutive_overlapping_subspans(span::AlignedSpan, duration::Period,
+                                     hop_duration::Period)
+    consecutive_overlapping_subspans(span::AlignedSpan, n::Int, m::Int)
```

Comment:
unless the variable `m` here has been selected for parity with some other pattern, I think we should rename it `hop_n` or something (and potentially even update `n` to `duration_n`, although I'm less sure/feel less strongly about that one). Reading code that deals with windows and sizes and hopping is confusing enough....